rusk/src/lib/http/rusk.rs

```diff
@@ -134,6 +140,33 @@ impl Rusk {
         let crs = rusk_profile::get_common_reference_string()?;
         Ok(ResponseData::new(crs).with_header("crs-hash", CRS_17_HASH))
     }
+
+    fn simulate_tx(&self, tx: &str) -> anyhow::Result<ResponseData> {
+        let tx = hex::decode(tx).map_err(|e| {
+            anyhow::anyhow!("Failed to decode transaction: {e}")
+        })?;
+        let tx = Transaction::from_slice(&tx)
+            .map_err(|e| anyhow::anyhow!("Invalid Data: {e:?}"))?;
+        if tx.gas_limit() > self.get_block_gas_limit() {
+            return Err(anyhow::anyhow!("Gas limit is too high."));
+        }
+        let mut session = self.query_session(None).map_err(|e| {
+            anyhow::anyhow!("Failed to initialize a session: {e:?}")
+        })?;
```

Comment:
This cannot be a query session
It has to be a `new_block_session` because you have to specify the block_height too